koku/api/report/aws/serializers.py

```diff
@@ -184,3 +198,125 @@ def validate_group_by(self, value):
                         }
                         raise serializers.ValidationError(error)
         return value
+
+
+class AWSEC2ComputeFilterSerializer(BaseFilterSerializer):
+    """Serializer for handling EC2 compute specific query parameter filter."""
+
+    RESOLUTION_CHOICES = (("monthly", "monthly"),)
+    TIME_CHOICES = (("-1", "-1"), ("-2", "-2"), ("-3", "-3"))
+    TIME_UNIT_CHOICES = (("month", "month"),)
+
+    _opfields = ("resource_id", "instance_name", "account", "operating_system", "region")
+
+    _aws_category = True
+
+    # override filtering with limit and offset params in the base `FilterSerializer` class.
+    # Not valid for this endpoint.
+    limit = None
+    offset = None
+
+    resource_id = StringOrListField(child=serializers.CharField(), required=False)
+    instance_name = StringOrListField(child=serializers.CharField(), required=False)
+    operating_system = StringOrListField(child=serializers.CharField(), required=False)
+    account = StringOrListField(child=serializers.CharField(), required=False)
+    region = StringOrListField(child=serializers.CharField(), required=False)
+
+    def validate(self, data):
+        """Validate incoming data.
+
+        Args:
+            data    (Dict): data to be validated
+        Returns:
+            (Dict): Validated data
+        Raises:
+            (ValidationError): if filter inputs are invalid
+
+        """
+        handle_invalid_fields(self, data)
+
+        resolution = data.get("resolution")
+        time_scope_value = data.get("time_scope_value")
+        time_scope_units = data.get("time_scope_units")
+
+        errors = {}
+
+        if not time_scope_units:
+            time_scope_units = TIME_SCOPE_UNITS_MONTHLY
+        if not time_scope_value:
+            time_scope_value = TIME_SCOPE_VALUES_MONTHLY[0]
+        if not resolution:
+            resolution = RESOLUTION_MONTHLY
+        if time_scope_units != TIME_SCOPE_UNITS_MONTHLY:
+            errors["time_scope_units"] = f"The valid value for time_scope_units is {TIME_SCOPE_UNITS_MONTHLY}."
+        if time_scope_value not in TIME_SCOPE_VALUES_MONTHLY:
+            errors["time_scope_value"] = f"The valid values for time_scope_value are {TIME_SCOPE_VALUES_MONTHLY}."
+        if resolution != RESOLUTION_MONTHLY:
+            errors["resolution"] = f"The valid value for resolution is {RESOLUTION_MONTHLY}."
+
+        if errors:
+            raise serializers.ValidationError(errors)
+
+        return data
```

Comment:
Because we are exclusively working with monthly units here, I think we can lean more heavily into the `ChoiceFields` logic instead of writing custom validation. We can override the `resolution`, `time_scope_value`, `time_scope_units` with:
```
    resolution = serializers.ChoiceField(
        choices=RESOLUTION_CHOICES,
        required=False,
        default=TIME_SCOPE_UNITS_MONTHLY,
        error_messages={"invalid_choice": f"valid choice is '{RESOLUTION_MONTHLY}'"},
    )
    time_scope_value = serializers.ChoiceField(
        choices=TIME_CHOICES,
        required=False,
        default=TIME_SCOPE_VALUES_MONTHLY[0],
        error_messages={"invalid_choice": f"valid choices are '{TIME_SCOPE_VALUES_MONTHLY}'"},
    )
    time_scope_units = serializers.ChoiceField(
        choices=TIME_UNIT_CHOICES,
        required=False,
        default=RESOLUTION_MONTHLY,
        error_messages={"invalid_choice": f"valid choice is '{TIME_SCOPE_UNITS_MONTHLY}'"},
    )
```

And then we can remove the validate method entirely.

We don't really have this luxury with the `FilterSerializer` because the daily vs monthly choices are exclusive which explains why `FilterSerializer` has that complicated validate method.

Reply:
Really should be letting serializer do the heavy lifting. Good catch, thanks!